proposals/texture-component-swizzle.md

```diff
@@ -21,29 +21,17 @@ partial enum GPUFeatureName {
 };
 ```
 
-The GPUTextureViewDescriptor `swizzle` option lets you rearrange or even replace the data from a texture's channels when creating a GPUTextureView. This is done by assigning a value from the GPUComponentSwizzle enum to each of the `r`, `g`, `b`, and `a` components in your texture view.
+A new `swizzle` option is added to the `GPUTextureViewDescriptor` which allows for rearranging or replacing the data from the texture's channels when creating a `GPUTextureView`. The `swizzle` value is a string of length four, with each character mapping to the view's red, green, blue, and alpha components, respectively. Each character can be either:
+- `"r"`: Take its value from the red channel of the texture.
+- `"g"`: Take its value from the green channel of the texture.
+- `"b"`: Take its value from the blue channel of the texture.
+- `"a"`: Take its value from the alpha channel of the texture.
+- `"0"`: Force its value to 0.
+- `"1"`: Force its value to 1.
 
 ```webidl
 partial dictionary GPUTextureViewDescriptor {
-    GPUTextureComponentSwizzle swizzle;
-};
-
-// Structure specifying a custom color component mapping for a texture view.
-dictionary GPUTextureComponentSwizzle {
-    GPUComponentSwizzle r = "r";
-    GPUComponentSwizzle g = "g";
-    GPUComponentSwizzle b = "b";
-    GPUComponentSwizzle a = "a";
-};
-
-// A set of options to choose from when specifying a component swizzle.
-enum GPUComponentSwizzle {
-    "zero",     // Force its value to 0.
-    "one",      // Force its value to 1.
-    "r",        // Take its value from the red channel of the texture.
-    "g",        // Take its value from the green channel of the texture.
-    "b",        // Take its value from the blue channel of the texture.
-    "a",        // Take its value from the alpha channel of the texture.
+    DOMString? swizzle;
```

Comment:
`DOMString swizzle = "rgba";`

and then we don't need handling of `null` in later parts.

Reply:
I think we still need the optional bit for C reasons. We pass only the swizzle descriptor to Dawn if swizzle is non-default because the C API will produce validation errors if a chained struct is passed without its feature being enabled. See https://chromium-review.googlesource.com/c/chromium/src/+/6958569

Reply:
You're right actually, I can get rid of this easily. I'll do that.